lib/charset.h

```diff
@@ -142,6 +142,8 @@ extern char *charset_unfold(const char *s, size_t len, int flags);
 
 extern int charset_decode(struct buf *dst, const char *src, size_t len, int encoding);
 
+extern int charset_decode_sha1(uint8_t *dest, size_t *outlen, const char *src, size_t len, int encoding);
```

Comment:
I find the name `outlen` confusing, I wondered why it is needed at all a 20-byte SHA1. Renaming it to `decodedlen` could avoid that confusion.